bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -199,10 +190,9 @@ pub mod pallet {
 			metadata: AssetMetadata,
 			fee: u128,
 		) -> DispatchResult {
-			let origin_location = match origin.clone().into() {
-				Ok(RawOrigin::Root) => Ok(Here.into()),
-				_ => T::SiblingOrigin::ensure_origin(origin),
-			}?;
+			let origin_location = T::SiblingOrigin::ensure_origin(origin)?;
+			let origin = AgentIdOf::convert_location(&origin_location)
```

Comment:
Why are you no reanchoring to be relative to Ethereum like in `create_agent`?

Reply:
> after reanchoring it will be (1,X2[GlobalConsensus(Polkadot),Parachain(1000)] which is incompatible with the agent for AH in V1.

Right, that is true. we need to factor out this "reanchoring and converting location -> hash" logic to a helper function. This helper function should have a match pattern that detects AH location, and uses V1 reanchoring instead of V2 reanchoring.

We can't duplicate this reanchoring/hashing logic in each governance method, it will introduce bugs.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/dbce06cf64ef54077ea725bac01fb490204b2ec8

Btw: since we introce `Descend(FrontendPallet)`, there is no chance conflict with V1.